narwhal/consensus/src/bullshark.rs

```diff
@@ -266,14 +266,27 @@ impl Bullshark {
     // Returns the PublicKey of the authority which is the leader for the provided `round`.
     // Pay attention that this method will return always the first authority as the leader
     // when used under a test environment.
-    pub fn leader_authority(committee: &Committee, _round: Round) -> PublicKey {
+    pub fn leader_authority(committee: &Committee, round: Round) -> PublicKey {
+        assert_eq!(
+            round % 2,
+            0,
+            "We should never attempt to do a leader election for odd rounds"
+        );
+
         cfg_if::cfg_if! {
             if #[cfg(test)] {
-                // consensus tests rely on returning the same leader.
-                committee.authorities.iter().next().expect("Empty authorities table!").0.clone()
+                // We apply round robin in leader election. Since we expect round to be an even number,
```

Comment:
Why do we want to use a different leader schedule in tests? My concern is that this is too different from the real behavior, i.e. stake weighted, same leader can be scheduled consecutively etc.

Reply:
I would say for integration tests we don't , but for unit tests where we need to have an expected outcome having a round robin helps as it is allowing us to easily check the outcome.

Reply:
Can we inject the leader schedule instead? Or can we use the real schedule in simtests and selected unit tests in future?

Reply:
I looked into injecting the leader schedule before as well and ultimately we want to do that - I'll do as separate refactoring. Now for simtests Vs tests we can still have the option of the weighted approach - I agree.

Reply:
Actually for the `simtests` the real schedule will be used - as those are tagged via a separate macro `sim_test`.